src/Workspaces/Core/Portable/Serialization/SerializerService.cs

```diff
@@ -225,12 +225,16 @@ void Serialize(string language, StructuredAnalyzerConfigOptions? options)
             return ImmutableDictionary<string, StructuredAnalyzerConfigOptions>.Empty;
         }
 
+        // We only serialize C# and VB options (if present):
+        Debug.Assert(count <= 2);
```

Comment:
nit.  consider making these strong COntract.Throw calls.  i think it's fine to throw in retail and force us to resolve asap if we break this.  (a lot of OOP syncing is done that way, and it has really helped squash issues).